schemas/x509.yml

```
@@ -52,8 +52,8 @@
       type: keyword
       short: Unique serial number issued by the certificate authority.
       description: >
-        Unique serial number issued by the certificate authority. For consistency, if this value is alphanumeric, it should be
-        formatted without colons and uppercase characters.
+        Unique serial number issued by the certificate authority. For consistency, this should be
+        encoded in base 16 and formatted without colons and uppercase characters.
```

"without colons and uppercase characters" is ambiguous. Does it mean "without colons and _without_ uppercase"? Or does it mean "without colons and _with_ uppercase"? 😄 

```suggestion
        Unique serial number issued by the certificate authority. For consistency, this should be
        encoded in base-16 and formatted as uppercase characters without colons.
```

---

Ha! Good point, my brain didn't spot that.
I think it should still be `base 16`, not `base-16`, as it's consistent with the other cases that use space, as in:
https://github.com/elastic/ecs/blob/main/schemas/dns.yml#L98
`grep -r 'base-'` doesn't return any other uses.

---

SGTM. If we haven't already created an `8.x` branch then we should do that _before_ you merge the second PR.

---

Agree, I'm just a bit confused, we have branches 8.{11,12,13,14}, when you say 8.x does this mean 8.{12,13,14} ?
Do we change ECS definitions (even if `should`), for current and older releases (<= 8.11).
In other words, where should I branch from?

---

This is what other repos like beats/kibana/elasticsearch and doing... We should create an `8.x` from `main`. And then future 8.{16,17,18} branches will come from `8.x`. We should backport to `8.x` unless the change is a something specifically for 9.0 (like a breaking change).

---

Got it, in the meantime @mjwolf volunteered to take care of this.

---

I've created an `8.x` branch now. I think the best way to handle this is to merge this PR to main first (and I'll handle the backport to the 8.x branch), and then afterwards create another PR with the changes intended for 9.x only that will stay in main only.

---

@haesbaert with this PR merged, and backported to 8.x, you can now create another PR with the changes intended for 9.x only. You can target it to `main`, and a `backport-skip` label